public/service-worker.js

```diff
@@ -0,0 +1,253 @@
+//https://developer.mozilla.org/en-US/docs/Web/API/Cache
+const OFFLINE_CACHE = "offline-cache-v1";
+const DATA_CACHE = "data-cache-v1";
+const NOTIFICATION_CACHE = "notification-cache-v1";
+const OFFLINE_URL = "offline.html";
+//Is localhost until I have an endpoint
+const ENDPOINT_URL = "localhost:3000";
+
+//These are the files downloaded into the cache for the PWA
+//If more offline pages are created for the PWA, they need to be added here
+const CACHE_STATIC_FILES = [
+  "/",
+  "/offline.html",
+  "/manifest.json",
+  "/static/fonts/montserrat.css",
+  "/logo192.png",
+  "/favicon.ico",
+];
+let sessionId = null;
+let notificationTimer = null;
+
+// This message is sent from index.html to provide the session id and timestamp to handle notifications.
+// event will have this data due to the index.html sending a postMessage with this data.
+// https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerGlobalScope/message_event
+self.addEventListener("message", async (event) => {
+  if (event.data.sessionId) {
+    sessionId = event.data.sessionId;
+    console.log("Session ID updated in service worker:", sessionId);
+  }
+  pollNotifications();
+});
+
+async function pollNotifications() {
+  // Will not start without a session ID or notification permissions
+  if (!sessionId || Notification.permission !== "granted") {
+    scheduleNextPoll();
+    return;
+  }
+  if (!(await canSendNotification())) {
+    console.log("Notification was recently shown. Skipping notification.");
+    scheduleNextPoll();
+    return;
+  }
+
+  try {
+    const data = await fetchNotificationData();
```

Comment:
In the real case you will need to check the "notification_available" property:

![{5BDC4379-703F-4C7F-93D4-DEF981870D33}](https://github.com/user-attachments/assets/2806db05-9363-46e1-9d1f-f884189748b7)

Ideally, I'd say you would want to mimic this to make sure the method would still run